### ISSUE TITLE
Environmental sensibilities

### ISSUE THREAD

**Issue by byorgey** [swarm-game/swarm#348]
This is pretty vague, but I've found myself troubled by the "strip mining" aspect of resource-gathering games like Minecraft and Factorio, in which certain resources are non-renewable, and the only way to play the game is to simply strip the world of these resources, searching father and farther afield as you use up the resources near you.  Factorio is particularly problematic, as you explicitly cause pollution which angers the native residents.  But it's OK to strip the resources of an area, pollute it, and kill the natives when they get upset, because, you know, they are evil non-human savages! 😠 😢 

Minecraft is a bit better --- there are renewable as well as non-renewable resources.  But there doesn't seem to be any way to avoid using up resources like iron, coal, etc.

I would like to make it at least *possible* to play Swarm in a more "environmentally sensitive" way, whatever that might mean, and want to brainstorm some ideas or directions we could take this.

- Of course we already do have some renewable resources e.g. trees, lambdas, and water.  We can certainly think about making more.
- Maybe we could make it possible to "recycle" things by breaking them down into their component ingredients.
- We could have alternate recipes for various things, or alternative devices giving certain capabilities, that are made using only renewable resources.  So e.g. if you want to play in such a way that you never use any nonrenewable resources it might be more difficult but possible (and perhaps there would also be some kind of benefit to doing so).

**Comment by xsebek** [comment]
We could also allow players to stick to renewables for the price of lower efficiency/speed.

For example, once we have iron and advanced recipes we could still allow players to make complex stuff from wooden gears. (DaVinci achievement?)

**Comment by byorgey** [comment]
Yes, and maybe there are other incentives to stick to renewables:

- Maybe it is less efficient/fast at first, but further down the tech tree maybe it leads to things which are actually *more* efficient.
- In relation to #218 , maybe your choices of what resources to use affects the attitude of native beings towards you.

**Comment by byorgey** [comment]
Maybe there could be some kind of advanced "fusion reactor" or "matter condenser" device which allows you to do things like e.g. make iron out of wood, you know, by rearranging the fundamental subatomic particles.

**Comment by byorgey** [comment]
I had a potentially interesting idea, which goes something like

- `furnace` + `copper pipes` + `water` -> `steam press`
- Using a `steam press`, you can turn 10 `board`s into one `ironwood board` (or something like that).  i.e. by pressure-treating the wood you can make a much stronger product which can be used as a substitute for iron in certain (not all) recipes.  So for some things you could either choose to go the mining route or the intensive forestry route.  We would have to figure out how to balance the two approaches.

**Comment by p3rsik** [comment]
Hey, I'm new here, but have an idea for this issue :) What do you think of robots having a battery which needs to be recharged? That way you would need to setup recharging station in the world(or maybe extend poles, so that bots would be "wireless"'ly charged. And if you want to integrate the renewable vs non-renewable resource theme here, you could think about where this energy would come from(for example, coal power plant vs solar panels with batteries). This will also allow for the scenarios that @byorgey wrote about in [here](https://github.com/swarm-game/swarm/issues/550#issuecomment-1299506729), where you would have limited resources(limited battery charge in robots) to do complex and challenging tasks.

**Comment by p3rsik** [comment]
@byorgey absolutely! We could take inspiration from Factorio bots here, where they go in the "slow" mode when the battery charge runs out. So thinking about energy can be only for the cases where you want "maximum perfomance" for the particular task, or something like that. Alternatively, there is a way of going with this whole "stages" idea, where you gradually develop technologies and unlock new capabilities. For example, installing a solar panel on the bot would allow it to operate in "perfomance" mode always, as opposed to bots that don't have it and must concern themselves with being energy-efficient.

I cannot talk about particular implementations yet, since I'm not yet familiar with the code enough, so I would skip that part of conversation, if you're okay with that.
